### DISCUSSION TITLE
Security when running on user-provided files?

### DISCUSSION THREAD

**Discussion by osbre** [gotenberg/gotenberg#931] [Q&A] [answered]
Hi. Say that you're a public web service and accepting uploads from users, passing them to this tool. Meanwhile, a new LibreOffice vulnerability gets detected/exists and therefore lets the attacker take over the container. Is that right?

If so, any plans to use this tool to spawn container instances with a timeout for each conversion for better security?

**Comment by gulien** [comment, marked as answer]
Hello @osbre,

Indeed, one could exploit LibreOffice vulnerabilities. However, we advise against running a public facing port.

>  If so, any plans to use this tool to spawn container instances with a timeout for each conversion for better security?

There is already a timeout per request.
